code-scanning/credo.yml

```diff
@@ -0,0 +1,60 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+
+# Credo is a static code analysis tool for the Elixir language with a focus on teaching and code consistency.
+# https://github.com/rrrene/credo
+#
+# To use this workflow, you must have GitHub Advanced Security (GHAS) enabled for your repository.
+#
+# Instructions:
+# 1. Add :credo as a dependency to your project's mix.exs with version ~> 1.7.0-rc.1 - https://github.com/rrrene/credo#installation-and-usage
+# 2. Follow the annotated workflow below and make any necessary modifications then save the workflow to your repository
+#    and review the "Security" tab once the action has run.
+
+name: Credo
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ $default-branch ]
+  schedule:
+    - cron: $cron-weekly
+
+permissions:
+  contents: read
+
+jobs:
+  security-scan:
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        otp: [version]
+        elixir: [version]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: erlef/setup-beam@v1
```

Comment:
This org doesn't seem to be part of Github Technology partners. Do we need this template or is it good to have?
Do we trust this org/repo that it would be maintained in future?

Reply:
Hi @Phantsure, thank you for pointing this out! I found this [PR](https://github.com/actions/starter-workflows/pull/822) and it looks like it's a supported org and is used in [ci/elixir](https://github.com/actions/starter-workflows/blob/d79edd9459bc72bd78ece2547c46cd0587e16d9e/ci/elixir.yml) . Would it be okay if I used the same commit hash here?

Reply:
> I found this https://github.com/actions/starter-workflows/pull/822 and it looks like it's a supported org and is used in [ci/elixir](https://github.com/actions/starter-workflows/blob/d79edd9459bc72bd78ece2547c46cd0587e16d9e/ci/elixir.yml)

Thanks for pointing this.

>  Would it be okay if I used the same commit hash here?

Yes. You can use any commit sha. Latest is preferred to have all new features.